internal/verifier/compare.go

```diff
@@ -118,6 +120,25 @@ func (verifier *Verifier) compareDocsFromChannels(
 	srcCache := map[string]docWithTs{}
 	dstCache := map[string]docWithTs{}
 
+	// NB: The below depends on strict, byte-level BSON equality. So if a
+	// document ID changes in some way the server considers equivalent, this
+	// will return 0. That’s OK, though, because the point of mismatch counting
+	// is to track the number of times a document was compared *without*
+	// a change.
+	var idToMismatch map[string]recheck.MismatchTimes
+	getPriorMismatchTimes := func(doc bson.Raw) option.Option[recheck.MismatchTimes] {
+		if idToMismatch == nil {
+			idToMismatch = createIdToMismatchTimes(task)
+		}
+
+		mapKeyBytes := rvToMapKey(
+			nil,
+			getDocIdFromComparison(verifier.docCompareMethod, doc),
+		)
+
+		return option.IfNotZero(idToMismatch[string(mapKeyBytes)])
+	}
```

Comment:
Does this need to be a closure? It seems like you could make this a separate func without too much trouble.